### DISCUSSION TITLE
ROOK-CEPH in EKS

### DISCUSSION THREAD

**Discussion by iflan7744** [rook/rook#10842] [Q&A]
I am running Rook-Ceph in EKS cluster with Auto scaler enabled. Therefore whenever there is a memory/Cpu issue the cluster scale in and scale out.

After new node is add or remove from the cluster the CEPH doesn't recognize it. 

Is there any script to run before and after creating/removing nodes therefore the ceph cluster will add the node to its pool.

**Comment by travisn** [comment]
If you are running in a dynamic environment like EKS, it's really recommended to follow the [cluster-on-pvc example](https://rook.io/docs/rook/latest/CRDs/Cluster/pvc-cluster/) with portable OSDs. This way if nodes are added or removed, the pods can just move to other nodes. 

If you really want to use the host cluster example to create OSDs on local devices, note that you may need additional [management of OSDs](https://rook.io/docs/rook/latest/Storage-Configuration/Advanced/ceph-osd-mgmt/#remove-an-osd) that may cause disruption when nodes are scaled down.

**Reply by iflan7744** [reply]
Let me try this and get back to you. Thank you @travisn
